recipe/meta.yaml

```diff
@@ -81,127 +90,84 @@ requirements:
-        - {{ compiler('c') }}                           # [not win]
-        - {{ compiler('cxx') }}                         # [not win]
-        - {{ compiler('fortran') }}                     # [not win]
-      run:   # [not win]
-        - {{ pin_subpackage('hdf5', exact=True) }}      # [not win]
-    script: install-static.sh  # [not win]
-    test:   # [not win]
-      commands:   # [not win]
-        {% set hdf5_libs = ["hdf5", "hdf5_cpp", "hdf5_hl", "hdf5_hl_cpp"] %}  # [not win]
-        {% for each_hdf5_lib in hdf5_libs %}            # [not win]
-        - test -f $PREFIX/lib/lib{{ each_hdf5_lib }}.a  # [not win]
-        {% endfor %}                                    # [not win]
+test:
+  requires:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}  # [not win]
+    - pkg-config                 # [not win]
+    # Need zlib to compile test programs
+    - zlib
+  files:
+    - h5_cmprss.c
+    - h5tutr_cmprss.cpp
+    - h5_cmprss.f90
+    - compound_fortran2003.f90
+  commands:
+    # Verify UNIX CLI tools.
+    {% if mpi == 'nompi' %}
+    {% set h5_compilers = [
+        "h5c++",
+        "h5cc",
+        "h5fc",
+    ] %}
+    {% else %}
+    # no p on parallel h5c++!
+    {% set h5_compilers = [
+        "h5c++",
+        "h5pcc",
+        "h5pfc",
+    ] %}
+    {% endif %}
+    {% set hdf5_unix_cmds = h5_compilers + [
+        "h5redeploy",
+    ] %}
+    {% if build_platform == target_platform %}
+    {% set hdf5_unix_cmds = hdf5_unix_cmds + [
+        "h5perf_serial",
+    ] %}
+    {% endif %}
+    {% for each_hdf5_unix_cmd in hdf5_unix_cmds %}
+    - command -v {{ each_hdf5_unix_cmd }}   # [not win]
+    {% endfor %}
+    # Verify CLI tools.
+    {% set hdf5_cmds = [
+        "gif2h5",
+        "h52gif",
+        "h5copy",
+        "h5debug",
+        "h5diff",
+        "h5dump",
+        "h5import",
+        "h5jam",
+        "h5ls",
+        "h5mkgrp",
+        "h5repack",
+        "h5repart",
+        "h5stat",
+        "h5unjam"
+    ] %}
+    {% for each_hdf5_cmd in hdf5_cmds %}
+    - command -v {{ each_hdf5_cmd }}  # [not win]
+    - where {{ each_hdf5_cmd }}  # [win]
+    {% endfor %}
+    # Verify libraries.
+    {% set hdf5_libs = [
+        "hdf5",
+        "hdf5_cpp",
+        "hdf5_hl",
+        "hdf5_hl_cpp"
+    ] %}
+    {% for each_hdf5_lib in hdf5_libs %}
+    - test -f $PREFIX/lib/lib{{ each_hdf5_lib }}${SHLIB_EXT}  # [not win]
+    - if not exist %PREFIX%\\Library\\lib\\{{ each_hdf5_lib }}.lib exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\bin\\{{ each_hdf5_lib }}.dll exit 1  # [win]
+    {% endfor %}
+    # Check ros3 driver
+    - h5dump --filedriver=ros3 "http://s3.amazonaws.com/hdfgroup/data/hdf5demo/tall.h5" | grep '^HDF5'  # [unix]
+    # Jan 2023 -- hmaarrfk
+    # I am really not sure why the ROS driver test doesn't pass on windows
+    # - h5dump --filedriver=ros3 "https://s3.amazonaws.com/hdfgroup/data/hdf5demo/tall.h5" | findstr '^HDF5'  # [win]
```

Comment:
@djhoese i had to skip the windows test here.

Reply:
Trying it locally, i also get 
```
h5dump error: unable to create FAPL for file access
```

Reply:
Yeah, i'm not sure, on windows the header has the line uncommented meaning that cmake thinks that it compiled with ROS3 support
```
#define H5_HAVE_ROS3_VFD 1
```

Reply:
I feel like I might be missing some context here. Regardless, is this something that should be filed as a bug/discussion upstream?

Reply:
done! they are on it! See link at the top level post!